### PR TITLE
[DRAFT] Switch namespace separator from colon to pipe

### PR DESCRIPTION
**Pull request by hartwork**

### CODE REVIEW

File: readme.txt [hunk 1]
```diff
@@ -266,8 +266,8 @@ RVP
        error ::= "error" patno erno error.
      z ::= "\0" .
 
-     * RVP assumes that the last colon in a name separates the local part
-       from the namespace URI (it is what one gets if specifies `:' as
+     * RVP assumes that a pipa in a name separates the local part
```
Comment (sydb) [inline readme.txt:269]: Typo: s/pipa/pipe/;
(I would just correct this myself but at the moment I cannot convince `git` to use my new PAT from GitHub.)

Comment (hartwork) [reply readme.txt]: Fixed, thanks :+1:

**Comment by bortzmeyer** [comment]
At this stage, while the rnv on this branch no longer has the "out of memory" problem, it has a lot of other complaints even with very simple schemas when they use namespaces:

```
schemas/./common.rnc:67:31: error: syntax error: "{" unexpected 
schemas/./common.rnc:70:18: error: syntax error: end of file expected, ")" found
...
schemas/fichelec.rnc:19:0: error: undefined reference to 'xsd:NMTOKEN'
schemas/fichelec.rnc:19:0: error: undefined reference to 'xsd:anyURI'
schemas/fichelec.rnc:19:0: error: undefined reference to 'xsd:integer'
schemas/fichelec.rnc:19:0: error: undefined reference to 'xsd:int'
schemas/fichelec.rnc:19:0: error: undefined reference to 'xsd:boolean'
```

**Comment by hartwork** [comment]
@bortzmeyer that matches what I heard from one other person. My contributions to RNV were [mostly on build system level](https://github.com/hartwork/rnv/commits?author=hartwork), and I don't know the C part of this codebase well, and the draft status is meant to communicate that I'm not confident about this status quo yet. Unfortunately the test suite doesn't detect breakage like that. If you (or a peer of yours) can help finding places that over-patch or under-patch colon in RNV, that would be welcome. (I'm also happy to jump on voice calls on the matter, if that helps for anything. My contact info is on my profile page.)

**Comment by hartwork** [comment]
PS: @bortzmeyer can you confirm that these issues only affect branch `namespace-separator` and that branch `master` does not show these issues?

**Comment by bortzmeyer** [comment]
> PS: @bortzmeyer can you confirm that these issues only affect branch `namespace-separator` and that branch `master` does not show these issues?

Branch master shows "just" the "out of memory" issue.

**Comment by hartwork** [comment]
> Branch master shows "just" the "out of memory" issue.

@bortzmeyer that makes sense. We'd need RNV master with Expat <2.4.5 or >=2.4.7 of course.

I'm closing this PR in favor of https://github.com/libexpat/libexpat/pull/577.

**Pull request closed without merging**